src/script.js

```diff
@@ -0,0 +1,298 @@
+function updateGameHostState(callFrame, isStartingGame) {
+  callFrame.setUserData({
+    isKahootHost: isStartingGame,
+  });
+
+  callFrame.sendAppMessage({ isGameStarted: isStartingGame }, '*');
+}
+
+/**
+ * Gets the current Kahoot host, if any, and returns their Daily session ID
+ * @param callFrame
+ * @returns {undefined|string}
+ */
+function getCurrentKahootHost(callFrame) {
+  const allParticipants = callFrame.participants();
+  const participants = Object.values(allParticipants);
+  for (let i = 0; i < participants.length; i += 1) {
+    const p = participants[i];
+    if (p.userData?.isKahootHost) {
+      return p.session_id;
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Enables the specified integrations (either Kahoot Host or Player),
+ * and disables the other one
+ * @param callFrame
+ * @param integration
+ */
+function toggleIntegrations(callFrame, integration) {
+  const integrations = callFrame.customIntegrations();
+  const customButtons = callFrame.customTrayButtons();
+  switch (integration) {
+    case Integration.Host: {
+      // Delete the Player integration and create a Host integration
+      removePlayerIntegration(integrations);
+      addHostIntegration(integrations, customButtons);
+      break;
+    }
+    case Integration.Player: {
+      // Delete theHost integration and create a Player integration
+      removeHostIntegration(integrations, customButtons);
+      addPlayerIntegration(integrations);
+      break;
+    }
+    default:
+      console.error('Unsupported integration requested: ', integration);
+      return;
+  }
+  // Now that all the integrations are constructed to the given mode (host or player),
+  // Have the call frame actually update its integrations as needed.
+  callFrame.setCustomIntegrations(integrations);
+  callFrame.updateCustomTrayButtons(customButtons);
+  if (integration === Integration.Player) {
+    // If the player integration is being activated, also start it.
+    callFrame.startCustomIntegrations([Integration.Player]);
+  }
+}
+
+/**
+ * Adds the player integration to the given integrations object
+ * @param integrations
+ */
+function addPlayerIntegration(integrations) {
+  const url = 'https://kahoot.it';
+  const playerIntegration = {
+    location: 'main',
+    shared: false,
+    src: url,
+    label: 'Play Kahoot',
+    sandbox: 'allow-same-origin allow-scripts allow-forms',
+  };
+  integrations[Integration.Player] = playerIntegration;
+}
+
+/**
+ * Removes the player integration from the given integrations object
+ * @param integrations
+ */
+function removePlayerIntegration(integrations) {
+  delete integrations[Integration.Player];
+}
+
+/**
+ * Adds the Kahoot Host integration to the given integrations object
+ * and adds the Kahoot host button to custom buttons
+ * @param integrations
+ * @param callFrame
+ */
+function addHostIntegration(integrations, customButtons) {
+  const hostIntegration = {
+    controlledBy: [],
+    location: 'main',
+    shared: true,
```

Comment:
I think this should be false based on what we agreed earlier.

Reply:
You are right!